python/pyarrow/parquet/__init__.py

```diff
@@ -3116,6 +3136,14 @@ def file_visitor(written_file):
                              "'basename_template' parameter instead. For "
                              "usage see `pyarrow.dataset.write_dataset`"),
             FutureWarning, stacklevel=2)
+    if existing_data_behavior is not None:
```

Comment:
```suggestion
    if existing_data_behavior != "overwrite_or_ignore":
```

? (since that is the default value. Or otherwise we need to change the default value to None and turn that into "overwrite_or_ignore" in the use_legacy_dataset=False code path)

Reply:
Oh, of course. Then I think it is better to set the default to None and set it in the new API separately to "overwrite_or_ignore". What do u think?

Reply:
Yes, sounds good